applications/OptimizationApplication/python_scripts/utilities/response_function_implementor.py

```diff
@@ -0,0 +1,194 @@
+
+    def GetSensitivity(self, sensitivity_variable, sensitivity_data_container: ContainerVariableDataHolderUnion, solution_step_index: int = 0):
+        key = f"problem_data/response_data/{sensitivity_data_container.GetModelPart().FullName()}/{self.GetName()}/sensitivities/{sensitivity_variable.Name()}/raw"
+        if solution_step_index == 0 and not self._optimization_info.HasValue(key):
+            # calculate the sensitivities
+            self.__response_function.CalculateSensitivity(sensitivity_variable, sensitivity_data_container.GetModelPart())
+
+            # transfer the model part variable data to ContainerVariableDataHolder object
+            sensitivity_data_container.ReadDataFromContainerVariable(sensitivity_variable)
+
+            # store it for future use if required.
+            self._optimization_info.SetValue(key, sensitivity_data_container.Clone())
+        else:
+            sensitivity_data_container.CopyDataFrom(self._optimization_info.GetValue(key, solution_step_index))
+
+    def _GetInitialResponseValue(self):
+        if self.__initial_response_value is None:
+            self.__initial_response_value = self.GetValue()
+
+        return self.__initial_response_value
+
+    @abstractmethod
+    def GetStandardizedValue(self, solution_step_index: int = 0) -> float:
+        pass
+
+    @abstractmethod
+    def GetStandardizedSensitivity(self, sensitivity_variable, sensitivity_data_container: ContainerVariableDataHolderUnion, solution_step_index: int = 0):
+        pass
+
+    @abstractmethod
+    def GetResponseType(self) -> str:
+        pass
+
+    @abstractmethod
+    def GetResponseInfo(self) -> str:
+        pass
+
+class ObjectiveResponseFunctionImplementor(ResponseFunctionImplementor):
+    def __init__(self, parameters: Kratos.Parameters, optimization_info: OptimizationInfo):
+        default_parameters = Kratos.Parameters("""{
+            "response_name": "",
+            "type"         : "",
+            "scaling"      : 1.0
+        }""")
+        parameters.ValidateAndAssignDefaults(default_parameters)
+
+        self.__scaling = parameters["scaling"].GetDouble()
+
+        self.__objective_type = parameters["type"].GetString()
+        if self.__objective_type == "minimization":
+            self.__scaling *= 1.0
+        elif self.__objective_type == "maximization":
+            self.__scaling *= -1.0
+        else:
+            raise RuntimeError(f"Requesting unsupported type {self.__objective_type} for objective response function. Supported types are: \n\tminimization\n\tmaximization")
+
+        super().__init__(parameters["response_name"].GetString(), optimization_info)
+
+    def GetStandardizedValue(self, solution_step_index: int = 0) -> float:
+        return self.GetValue(solution_step_index) * self.__scaling
+
+    def GetStandardizedSensitivity(self, sensitivity_variable, sensitivity_data_container: ContainerVariableDataHolderUnion, solution_step_index: int = 0):
```

Comment:
The same applies here

Reply:
It should be called, StandardizeSensitivity()

Reply:
`CalculateStandardizedSensitivity`

Reply:
done :)